args.go

```diff
@@ -81,6 +82,43 @@ func getGlobalArgs() (*logger.GlobalArgs, error) {
 	return args, nil
 }
 
+// getDockerConfigs gets the optional docker config variables
+func getDockerConfigs() (*logger.DockerConfigs, error) {
+	containerLabelsString := viper.GetString(ContainerLabelsKey)
+	containerLabels := make(map[string]string)
+	if containerLabelsString != "" {
+		err := json.Unmarshal([]byte(containerLabelsString), &containerLabels)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	containerEnvString := viper.GetString(ContainerEnvKey)
+	containerEnv := make([]string, 0)
+	if containerEnvString != "" {
+		containerEnvMap := make(map[string]string)
+		err := json.Unmarshal([]byte(containerEnvString), &containerEnvMap)
+		if err != nil {
+			return nil, err
+		}
+		// Docker logging use a slice to store the environment variables
+		// Each item is in the format of "key=value"
+		// ref: https://github.com/moby/moby/blob/c833222d54c00d64a0fc44c561a5973ecd414053/daemon/logger/loginfo.go#L60
+		for envKey, envVal := range containerEnvMap {
+			containerEnv = append(containerEnv, envKey+"="+envVal)
+		}
+	}
+
+	args := &logger.DockerConfigs{
+		ContainerImageID:   viper.GetString(ContainerImageIDKey),
```

Comment:
What would be the behavior here if container image ID and name are empty?

Reply:
We will pass empty values to docker logger, similar to all other optional arguments. There is a unit test `testGetDockerConfigsEmpty` covering this case.

Reply:
Can you add a documentation here in case we lost tack of the explanation?

Reply:
Sure will do.